scripts/filter_multimappers.py

```diff
@@ -133,13 +134,20 @@ def main(sam_file: Path) -> None:
         sam_file: Path to the input SAM file.
     """
     with pysam.AlignmentFile(sam_file, "r") as samfile:
+      
+        try:
+            first_alignment = next(samfile)
+            
+        except StopIteration:
+            sys.stdout.write(str(samfile.header))
+            return
```

Comment:
What's the point of this code? Is it to take care of SAM files without any alignments? Wouldn't this just be automatically taken care by the loop (which would have zero iterations)?

Reply:
In case of a SAM file only containing the header, when the script is ran, it does it with "empty alignments" therefore it raises an error.

Reply:
What's the error, and where is it raised? Generally, when you have a loop that has nothing to iterate over (an empty list, dictionary, or file), the loop is simply not executed.

Is this just coming from the `write_output()` that is outside of the loop? If so, that's easy to avoid - just check whether `current_alignments` or `current_query` are empty. That's much more simple/elegant than handling the first alignment separately.